tests/phpunit/CRM/Civicase/Service/CaseCategoryInstanceTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+use CRM_Civicase_Service_CaseCategoryInstance as CaseCategoryInstanceService;
+use CRM_Civicase_Test_Fabricator_CaseCategoryInstance as CaseCategoryInstanceFabricator;
+use CRM_Civicase_Test_Fabricator_CaseCategory as CaseCategoryFabricator;
+
+/**
+ * Runs tests on CaseCategoryInstance Service tests.
+ *
+ * @group headless
+ */
+class CRM_Civicase_Service_CaseCategoryInstanceTest extends BaseHeadlessTest {
+
+  /**
+   * Test that the category instances list is correctly formatted.
+   */
+  public function testGetCategoryInstances() {
+    // Generate random category instances.
+    $categoryInstanceOne = CaseCategoryInstanceFabricator::fabricate();
+    $categoryInstanceTwo = CaseCategoryInstanceFabricator::fabricate();
+    $caseCategoryInstanceService = new CaseCategoryInstanceService();
+
+    // We call the method for getting the list.
+    $caseCategoryInstances = $caseCategoryInstanceService->getCaseCategoryInstances();
+
+    // And we verify the structure and content of the list received.
+    $this->assertArrayHasKey($categoryInstanceOne['id'], $caseCategoryInstances);
+    $this->assertArrayHasKey($categoryInstanceTwo['id'], $caseCategoryInstances);
+    $this->assertEquals(
+      $categoryInstanceOne['category_id'],
+      $caseCategoryInstances[$categoryInstanceOne['id']]->category_id
+    );
+    $this->assertEquals(
+      $categoryInstanceOne['instance_id'],
+      $caseCategoryInstances[$categoryInstanceOne['id']]->instance_id
+    );
+    $this->assertEquals(
+      $categoryInstanceTwo['category_id'],
+      $caseCategoryInstances[$categoryInstanceTwo['id']]->category_id
+    );
+    $this->assertEquals(
+      $categoryInstanceTwo['instance_id'],
+      $caseCategoryInstances[$categoryInstanceTwo['id']]->instance_id
+    );
+  }
+
+  /**
+   * Test that the category instances list by name is correctly formatted.
+   */
+  public function testGetCategoryInstancesByInstanceName() {
+    // Generate random category instances.
+    CaseCategoryInstanceFabricator::fabricate();
+    CaseCategoryInstanceFabricator::fabricate();
+    // And get a random instance type.
+    $categoryInstanceType = $this->getRandomCategoryInstanceType();
+    $caseCategoryInstanceService = new CaseCategoryInstanceService();
+
+    // We call the method for getting the list.
+    $caseCategoryInstances = $caseCategoryInstanceService->getCaseCategoryInstances($categoryInstanceType['name']);
+
+    // And check that all correspond to the instance type specified.
+    foreach ($caseCategoryInstances as $categoryInstance) {
+      $this->assertEquals($categoryInstanceType['value'], $categoryInstance->instance_id);
+    }
+  }
+
+  /**
+   * Test the correct creation of an instance for an existing category.
+   */
+  public function testAssignInstanceForExistingCaseCategories() {
```

Comment:
Can we rename to `testAssignInstanceForExistingCaseCategoriesWithoutAnInstance`

Reply:
@tunbola Done.